core/src/main/java/org/nervos/ckb/sign/signer/AcpSigner.java

```diff
@@ -0,0 +1,39 @@
+package org.nervos.ckb.sign.signer;
+
+import org.nervos.ckb.sign.Context;
+import org.nervos.ckb.sign.ScriptGroup;
+import org.nervos.ckb.sign.ScriptSigner;
+import org.nervos.ckb.type.Script;
+import org.nervos.ckb.type.transaction.Transaction;
+import org.nervos.ckb.utils.Numeric;
+
+public class AcpSigner implements ScriptSigner {
+  private Secp256k1Blake160SighashAllSigner secp256K1Blake160SighashAllSigner =
+      Secp256k1Blake160SighashAllSigner.getINSTANCE();
+  private static AcpSigner INSTANCE;
+
+  private AcpSigner() {}
+
+  public static AcpSigner getINSTANCE() {
+    if (INSTANCE == null) {
+      INSTANCE = new AcpSigner();
+    }
+    return INSTANCE;
+  }
+
+  @Override
+  public boolean signTransaction(
+      Transaction transaction, ScriptGroup scriptGroup, Context context) {
+    Script script = scriptGroup.getScript();
+    String privateKey = context.getPrivateKey();
+    if (isMatched(privateKey, script.args) == false) {
+      return false;
+    }
+    return secp256K1Blake160SighashAllSigner.unlockScript(transaction, scriptGroup, privateKey);
+  }
+
+  public boolean isMatched(String privateKey, String scriptArgs) {
+    return secp256K1Blake160SighashAllSigner.isMatched(
+        privateKey, Numeric.cleanHexPrefix(scriptArgs).substring(0, 40));
+  }
```

Comment:
Acp script args can be 20/21/22 bytes
* 20 => only pubkey hash
* 21 => pubkey hash + min_ckb_amount
* 22 => pubkey hash + min_ckb_amount + min_udt_amount